Mathlib/Algebra/Star/Defs.lean

```diff
@@ -0,0 +1,179 @@
+/-
+Copyright (c) 2020 Kim Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kim Morrison
+-/
+
+import Mathlib.Data.SetLike.Basic
+
+#align_import algebra.star.basic from "leanprover-community/mathlib"@"31c24aa72e7b3e5ed97a8412470e904f82b81004"
+
+/-!
+# Star monoids, rings, and modules
+
+We introduce the basic algebraic notions of star monoids, star rings, and star modules.
+A star algebra is simply a star ring that is also a star module.
```

Comment:
Not all of this is introduced in the `Defs` file. How did you decide where to make this split? It seems like, if we can, `StarAddMonoid` and `StarRing` should go in here too.